keras_nlp/src/models/mobilenet/mobilenet_backbone.py

```diff
@@ -0,0 +1,551 @@
+        include_rescaling,
+        output_filter,
+        activation,
+        inverted_res_block,
+        depth_multiplier=1.0,
+        input_filter=16,
+        image_shape=(224, 224, 3),
+        **kwargs,
+    ):
+        activation_str = activation
+        if isinstance(activation, str):
+            if activation == "hard_swish":
+                activation = apply_hard_swish
+            elif activation == "relu6":
+                activation = apply_relu6
+            else:
+                activation = keras.activations.get(activation)
+
+        # === Functional Model ===
+
+        inputs = keras.layers.Input(shape=image_shape)
+        x = inputs
+
+        if include_rescaling:
+            x = keras.layers.Rescaling(scale=1 / 255)(x)
+
+        input_filter = adjust_channels(input_filter)
+        x = keras.layers.Conv2D(
+            input_filter,
+            kernel_size=3,
+            strides=(2, 2),
+            padding="same",
+            use_bias=False,
+            name="Conv",
+        )(x)
+        x = keras.layers.BatchNormalization(
+            axis=CHANNEL_AXIS,
+            epsilon=BN_EPSILON,
+            momentum=BN_MOMENTUM,
+            name="Conv_BatchNorm",
+        )(x)
+
+        x = activation(x)
+
+        for stack_index in range(len(stackwise_filters)):
+
+            if inverted_res_block:
+                x = apply_inverted_res_block(
+                    x,
+                    expansion=stackwise_expansion[stack_index],
+                    filters=adjust_channels(
+                        (stackwise_filters[stack_index]) * depth_multiplier
+                    ),
+                    kernel_size=stackwise_kernel_size[stack_index],
+                    stride=stackwise_stride[stack_index],
+                    se_ratio=(
+                        stackwise_se_ratio[stack_index]
+                        if activation_str == "hard_swish"
+                        else 0
+                    ),
+                    activation=stackwise_activation[stack_index],
+                    expansion_index=stack_index,
+                )
+            else:
+                x = apply_depthwise_conv_block(
+                    x,
+                    filters=adjust_channels(
+                        (stackwise_filters[stack_index]) * depth_multiplier
+                    ),
+                    kernel_size=3,
+                    stride=stackwise_stride[stack_index],
+                    depth_multiplier=depth_multiplier,
+                    block_id=stack_index,
+                )
+
+        if output_filter is not None:
+            if activation_str == "hard_swish":
+                last_conv_ch = adjust_channels(x.shape[CHANNEL_AXIS] * 6)
+            else:
+                last_conv_ch = output_filter
+
+            x = keras.layers.Conv2D(
+                last_conv_ch,
+                kernel_size=1,
+                padding="same",
+                use_bias=False,
+                name="Conv_1",
```

Comment:
why are these `_1`, seems like a strange suffix. maybe name this `output_conv` and `output_batch_norm` and the input ones `input_conv` and `input_batch_norm`?